paddle/fluid/framework/operator.h

```diff
@@ -128,6 +129,9 @@ class OperatorBase {
   //! Get all outputs variable names
   virtual std::vector<std::string> OutputVars(bool has_intermediate) const;
 
+  virtual void RunInferShape(const Scope& scope,
```

Comment:
Why need add this interface? we already have 
```
  RuntimeInferShapeContext infer_shape_ctx(*this, scope);
  this->InferShape(&infer_shape_ctx);
```

Reply:
This is to avoid exposing the RuntimeInferShapeContext class. It is more like to choice to make.